### PR TITLE
feat(components): create sign-up page

### LINKED ISSUE #19: Create sign-up page

### PR DESCRIPTION
**Pull request by Hellomyguest**
META DATA: issue19

### CODE REVIEW

File: src/pages/sign/signUp/model.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { createEvent, createStore } from "effector";
+
+type User = {
+  nickName: string;
+  name: string;
+  surname: string;
+  patronymic: string;
+  password: string;
+  isAdmin: boolean;
```
Comment (makhnatkin) [inline src/pages/sign/signUp/model.ts]: delete `isAdmin: boolean;`
use Admins array 
```
type User = {
  id: string;
  nickName: string;
  name: string;
  surname: string;
  patronymic: string;
  password: string;
}
type AdminsIds = string[];
```

File: src/pages/sign/signUp/model.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { createEvent, createStore } from "effector";
+
+type User = {
+  nickName: string;
+  name: string;
+  surname: string;
+  patronymic: string;
```
Comment (makhnatkin) [inline src/pages/sign/signUp/model.ts]:
```
patronymic?: string;
```

File: src/pages/sign/signUp/model.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  isAdmin: boolean;
+};
+export const $users = createStore<User[]>([
+  {
+    nickName: "admin",
+    password: "123",
+    name: "",
+    surname: "",
+    patronymic: "",
+    isAdmin: true,
```
Comment (makhnatkin) [inline src/pages/sign/signUp/model.ts]: delete isAdmin: true,

File: src/pages/sign/signUp/page.css [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+.Content {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.Form {
+  width: 15rem;
```
Comment (makhnatkin) [inline src/pages/sign/signUp/page.css]: 100%

File: src/pages/sign/signUp/page.tsx [hunk 5]
```diff
@@ -1,11 +1,169 @@
...
+};
+
+export const SignUpPage = () => {
+  const [users, addUserFn] = useUnit([$users, addUser]);
+
+  const onFinish = (values: User) => {
+    const user = { ...values, isAdmin: false };
+    delete user.confirm;
+    addUserFn(user);
+    // eslint-disable-next-line effector/mandatory-scope-binding
```
Comment (makhnatkin) [inline src/pages/sign/signUp/page.tsx]: dont disable lines! its very bad practise

File: src/pages/sign/signUp/page.tsx [hunk 6]
```diff
@@ -1,11 +1,169 @@
...
+  confirm: never;
+};
+
+export const SignUpPage = () => {
+  const [users, addUserFn] = useUnit([$users, addUser]);
+
+  const onFinish = (values: User) => {
+    const user = { ...values, isAdmin: false };
+    delete user.confirm;
+    addUserFn(user);
```
Comment (makhnatkin) [inline src/pages/sign/signUp/page.tsx]: fn - delete

File: src/pages/sign/signUp/page.tsx [hunk 7]
```diff
@@ -1,11 +1,169 @@
...
+          className="Form"
+          labelCol={{ span: 24 }}
+          wrapperCol={{ span: 24 }}
+          name="register"
+          onFinish={onFinish}
+          scrollToFirstError
+        >
+          <Form.Item
+            name="nickName"
+            rules={[
```
Comment (makhnatkin) [inline src/pages/sign/signUp/page.tsx:50]: rules to other file

File: src/pages/sign/signUp/page.tsx [hunk 8]
```diff
@@ -1,11 +1,169 @@
...
+                whitespace: true,
+              },
+              () => ({
+                validator(_, value) {
+                  if (users.every((user) => user.nickName !== value)) {
+                    return Promise.resolve();
+                  }
+                  return Promise.reject(new Error("Такой ник занят!"));
+                },
+              }),
```
Comment (makhnatkin) [inline src/pages/sign/signUp/page.tsx:63]: code duplication in two prs

File: src/pages/sign/signUp/model.ts (hunk 1 above)
Comment (Hellomyguest) [reply src/pages/sign/signUp/model.ts]: done

File: src/pages/sign/signUp/model.ts (hunk 2 above)
Comment (Hellomyguest) [reply src/pages/sign/signUp/model.ts]: done

File: src/pages/sign/signUp/model.ts (hunk 3 above)
Comment (Hellomyguest) [reply src/pages/sign/signUp/model.ts]: done

File: src/pages/sign/signUp/page.tsx (hunk 6 above)
Comment (Hellomyguest) [reply src/pages/sign/signUp/page.tsx]: done

File: src/pages/sign/signUp/page.tsx (hunk 5 above)
Comment (Hellomyguest) [reply src/pages/sign/signUp/page.tsx]: in last commit it was deleted

File: src/pages/sign/signUp/page.tsx [hunk 9]
```diff
@@ -1,11 +1,174 @@
...
+export const SignUpPage = () => {
+  const [users, addUserFn, signupFormSubmittedFn] = useUnit([
+    $users,
+    addUser,
+    signupFormSubmitted,
+  ]);
+
+  const onFinish = (values: User) => {
+    addUserFn(deleteConfirmAddId(values));
+    signupFormSubmittedFn();
```
Comment (dpolevodin) [inline src/pages/sign/signUp/page.tsx:34]: I think that you can not call signupFormSubmittedFn in the handler, can you also transfer this to the effector model?
eg:
```
redirect({
  clock: addUserFn,
  route: routes.student,
});
```

File: src/pages/sign/signUp/model.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  {
+    id: "f115a395-19b4-0b15-f474-33dbf9bd9e3b",
+    nickName: "admin",
+    password: "123",
+    name: "",
+    surname: "",
+  },
+]);
+
+export const $adminIds = createStore<AdminIds>([
```
Comment (makhnatkin) [inline src/pages/sign/signUp/model.ts:26]: add TODO comment

**Review by makhnatkin — APPROVED** [review]

**Review by dpolevodin — APPROVED** [review]

**Pull request merged**